app/helpers/ci_status_helper.rb

```diff
@@ -51,4 +51,8 @@ def render_ci_status(ci_commit)
       ci_status_icon(ci_commit)
     end
   end
+
+  def ci_widget_summary(commit)
+    "#{commit.statuses_with_matching_status} of #{commit.latest_generic_statuses.length + 1} checks #{commit.human_status}"
```

Comment:
> Why distinguish generic statuses from Ci::Builds? The builds are also "checks", but for different test suites.

I'll use the TravisCI terminology of builds which contain many jobs. I do not think that we should list the status of every job but rather the build status. Showing details of the build (the individual jobs) should be the responsibility of the service (GitlabCI, TravisCI, Coveralls, etc) that reported the status.

> Also this method is crazy - no idea what is happening here ;)

Don't know what is so crazy about it, but of course are open for simplification suggestions. 
`commit.statuses_with_matching_status` returns the amount of statuses that match the current status of the MR itself. 
`commit.latest_generic_statuses.length` is the amount of generic statuses, `+ 1` because no matter how many CI builds there are they are always grouped into one status. 
`commit.human_status` changes `success` to `succeeded` because "4 of 4 checks success" would not be a real sentence.